src/modules/flight_mode_manager/tasks/Utility/StickTiltXY.cpp

```diff
@@ -59,6 +64,26 @@ Vector2f StickTiltXY::generateAccelerationSetpoints(Vector2f stick_xy, const flo
 	Sticks::limitStickUnitLengthXY(stick_xy);
 	_man_input_filter.setParameters(dt, _param_mc_man_tilt_tau.get());
 	stick_xy = _man_input_filter.update(stick_xy);
+
 	Sticks::rotateIntoHeadingFrameXY(stick_xy, yaw, yaw_setpoint);
+
 	return stick_xy * _maximum_acceleration;
 }
+
+Vector2f StickTiltXY::generateAccelerationSetpointsForAltitudeCruise(Vector2f stick_xy, const float dt, const float yaw,
+		const float yaw_setpoint)
+{
+	Sticks::limitStickUnitLengthXY(stick_xy);
+	const Vector2f increment = stick_xy;
+	_altitude_cruise_acceleration += increment * _maximum_acceleration * 2.f *
+					 dt; // at full stick deflection it takes 1s from -tilt_max to tilt_max
+
+	if (_altitude_cruise_acceleration.longerThan(_maximum_acceleration)) {
+		_altitude_cruise_acceleration =
+			_altitude_cruise_acceleration.unit_or_zero() * _maximum_acceleration;
+	}
+
+	auto global_altitude_cruise_acceleration = _altitude_cruise_acceleration;
```

Comment:
Static type checking -> write the type -> find mistakes early
```suggestion
	Vector2f global_altitude_cruise_acceleration = _altitude_cruise_acceleration;
```